cve_bin_tool/cvedb.py

```diff
@@ -865,7 +865,7 @@ def get_all_records_in_table(self, table_name):
         """Return JSON of all records in a database table."""
         cursor = self.db_open_and_get_cursor()
         cursor.row_factory = self.dict_factory
-        cursor.execute(f"SELECT * FROM '{table_name}' ")
+        cursor.execute(f"SELECT * FROM '{table_name}' ") #nosec
```

Comment:
I'm not actually 100% this will work.  I don't think ? works correctly with tablenames.  In which case you'll likely have to do something similar to what I mention below.

Reply:
For this what if we create a sanatize function:
-The `sanitize_table_name()` function checks if the table name contains only alphanumeric characters and underscores. -If it contains any other characters, it raises a `ValueError`
-Before constructing the SQL query, the `table_name` is sanitized using the `sanitize_table_name()` function.
-The sanitized `table_name` is then interpolated into the SQL query string as indicated below:

```python
def sanitize_table_name(table_name):
    # Check if the table name contains only valid characters
    if not table_name.isalnum() and "_" not in table_name:
        raise ValueError("Invalid table name")
    
    # Return the sanitized table name
    return table_name
 def get_all_records_in_table(self, table_name):
    """Return JSON of all records in a database table."""
    # Sanitize the table name to prevent injection
    table_name = sanitize_table_name(table_name)

    cursor = self.db_open_and_get_cursor()
    cursor.row_factory = self.dict_factory

    # Construct the SQL query with the sanitized table name
    query = f"SELECT * FROM '{table_name}' "
    ```